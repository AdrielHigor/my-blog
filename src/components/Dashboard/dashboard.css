div#Dashboard {
    width: 100%;
    max-width: 450px;
    margin: 30px auto 0;
    padding: 30px;
    background: #ffffff;
    border: 1px solid #dddddd;
    display: flex;
    flex-direction: column;
}

div#Dashboard h1{
    text-align: center;
}

form#Dashboard p{
    text-align: center;
    text-decoration: none;
    font-size: 16px;
    padding-top: 20px;
    margin-bottom: 15px;
}

div#Dashboard button{
    padding:10px 20px;
    border-radius: 4px;
    background: #ff0000;
    color: #ffffff;
    font-size: 18px;
    font-weight: bold;
    cursor: pointer;
    margin-bottom: 5px;
}

a#newpost {
    width: 50%;
    text-align: center;
    padding:10px 20px;
    border-radius: 4px;
    background: #25282a;
    color: #ffffff;
    font-size: 18px;
    font-weight: bold;
    cursor: pointer;
    margin: 10px auto;
}