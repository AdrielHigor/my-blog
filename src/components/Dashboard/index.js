import React, { useEffect } from "react";
import { Link, withRouter } from "react-router-dom";
import firebase from "../../firebase";
import "./dashboard.css";

export default withRouter(function Dashboard(props) {
    const [name, setName] = React.useState([]);
    const [role, setRole] = React.useState([]);
    const [email, setEmail] = React.useState([]);

    const logout = async () => {
        await firebase.logout();
        props.history.replace("/");
    };

    useEffect(() => {
        if (!firebase.getCurrent()) {
            props.history.replace("/login");
        }

        setEmail(firebase.getCurrent());

        firebase.getUserInfo(info => {
            setName(info.val().name);
            setRole(info.val().role);
        });
    });

    return (
        <div id="Dashboard">
            <div>
                <h1>Ol√° {name}</h1>
                <p>Email: {email}</p>
            </div>
            {role === "admin" ?
                <Link id="newpost" to="/dashboard/newpost">
                    Novo Post
                </Link> : null}
            <button onClick={logout}>Sair</button>
        </div>
    )
});
