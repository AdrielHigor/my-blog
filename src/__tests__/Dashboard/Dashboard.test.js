import React from "react";
import Dashboard from "../../components/Dashboard";
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from "enzyme";
import { Link } from "react-router-dom";

configure({ adapter: new Adapter() });

describe("Dashboard Test Suite", () => {
    test("Dashboard rendering correctly", () => {
        let component

        jest
            .spyOn(React, 'useState')
            .mockImplementationOnce(() => ["Test User", jest.fn()])
            .mockImplementationOnce(() => ["user", jest.fn()])
            .mockImplementationOnce(() => ["testemail@teste.com", jest.fn()])

        component = shallow(<Dashboard.WrappedComponent />)

        expect(component.text()).toContain("testemail@teste.com")
        expect(component.text()).toContain("Test User")

        expect(component.html()).toMatchSnapshot();
    })

    test("Dashboard showing add post for admin", () => {
        let component

        jest
            .spyOn(React, 'useState')
            .mockImplementationOnce(() => ["Test Admin", jest.fn()])
            .mockImplementationOnce(() => ["admin", jest.fn()])
            .mockImplementationOnce(() => ["testemail@teste.com", jest.fn()])

        component = shallow(<Dashboard.WrappedComponent />)

        expect(component.find(Link).length).toEqual(1);
    })

    test("Dashboard hiding add post for normal user", () => {
        let component

        jest
            .spyOn(React, 'useState')
            .mockImplementationOnce(() => ["Test User", jest.fn()])
            .mockImplementationOnce(() => ["user", jest.fn()])
            .mockImplementationOnce(() => ["testemail@teste.com", jest.fn()])

        component = shallow(<Dashboard.WrappedComponent />)

        expect(component.find(Link).length).toEqual(0);
    })
})