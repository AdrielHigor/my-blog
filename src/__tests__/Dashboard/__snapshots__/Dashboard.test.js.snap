// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Test Suite Dashboard rendering correctly 1`] = `"<div id=\\"Dashboard\\"><div><h1>Ol√° Test User</h1><p>Email: testemail@teste.com</p></div><button>Sair</button></div>"`;
